### PR TITLE
Fixes #32 - Reschedules schema job upon failure

### LINKED ISSUE #32: Reschedule `jaeger-cassandra-schema-job` upon failure
I'm using the production YAML file to deploy Jager : 

```
root@kmaster01:~# kubectl  --namespace=opentracing get pod
NAME                                READY     STATUS             RESTARTS   AGE
cassandra-0                         1/1       Running            0          5m
cassandra-1                         1/1       Running            1          5m
cassandra-2                         1/1       Running            3          5m
jaeger-collector-1578691591-gm9dn   0/1       CrashLoopBackOff   3          6m
jaeger-query-1666845303-113hb       0/1       CrashLoopBackOff   2          6m

root@kmaster01:~# kubectl  --namespace=opentracing logs jaeger-collector-1578691591-gm9dn
2017/09/06 14:42:56 error: failed to connect to 10.244.48.54:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:42:56 error: failed to connect to 10.244.80.25:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:42:56 error: failed to connect to 10.244.128.36:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:42:56 error: failed to connect to 10.244.48.54:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:42:56 error: failed to connect to 10.244.80.25:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:42:57 error: failed to connect to 10.244.128.36:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
{"level":"fatal","ts":1504708977.0914874,"caller":"collector/main.go:79","msg":"Unable to set up builder","error":"no connections were made when creating the session","stacktrace":"github.com/uber/jaeger/vendor/go.uber.org/zap.Stack\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/field.go:209\ngithub.com/uber/jaeger/vendor/go.uber.org/zap.(*Logger).check\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/logger.go:273\ngithub.com/uber/jaeger/vendor/go.uber.org/zap.(*Logger).Fatal\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/logger.go:208\nmain.main.func1\n\t/home/travis/gopath/src/github.com/uber/jaeger/cmd/collector/main.go:79\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).execute\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:636\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:722\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).Execute\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:681\nmain.main\n\t/home/travis/gopath/src/github.com/uber/jaeger/cmd/collector/main.go:122"}

root@kmaster01:~# kubectl  --namespace=opentracing logs jaeger-query-1666845303-113hb
2017/09/06 14:43:12 error: failed to connect to 10.244.128.36:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:43:12 error: failed to connect to 10.244.80.25:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:43:12 error: failed to connect to 10.244.48.54:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:43:12 error: failed to connect to 10.244.128.36:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:43:12 error: failed to connect to 10.244.80.25:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
2017/09/06 14:43:12 error: failed to connect to 10.244.48.54:9042 due to error: Keyspace 'jaeger_v1_dc1' does not exist
{"level":"fatal","ts":1504708993.0407972,"caller":"query/main.go:71","msg":"Failed to init storage builder","error":"no connections were made when creating the session","stacktrace":"github.com/uber/jaeger/vendor/go.uber.org/zap.Stack\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/field.go:209\ngithub.com/uber/jaeger/vendor/go.uber.org/zap.(*Logger).check\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/logger.go:273\ngithub.com/uber/jaeger/vendor/go.uber.org/zap.(*Logger).Fatal\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/logger.go:208\nmain.main.func1\n\t/home/travis/gopath/src/github.com/uber/jaeger/cmd/query/main.go:71\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).execute\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:636\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:722\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).Execute\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:681\nmain.main\n\t/home/travis/gopath/src/github.com/uber/jaeger/cmd/query/main.go:100"}
```

Any idea why i've got these errors ? Thanks.

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: production/jaeger-production-template.yml [hunk 1]
```diff
@@ -120,13 +120,13 @@ items:
       spec:
         containers:
         - name: jaeger-cassandra-schema
-          image: jaegertracing/jaeger-cassandra-schema:0.6
+          image: jaegertracing/jaeger-cassandra-schema:0.8
```
Comment (pavolloffay) [inline production/jaeger-production-template.yml]: if you are updating versions in this PR could you update also all-in-one?

**Comment by jpkrohling** [comment]
Besides the rescheduling, this brings the 0.8 version, which should have been done as part of #38 .

@kevinearls , would you test this one and perhaps add this to the test plan?

Test:

1) Install the production template without Cassandra
2) Wait about 1 minute

Current outcome: the create-schema job gets into a `Error` state (not sure why there are two pods for this job there)
image

New, expected outcome: the create-schema job is rescheduled and verify that the number of restarts is bigger than 0.
image

File: production/jaeger-production-template.yml (hunk 1 above)
Comment (jpkrohling) [reply production/jaeger-production-template.yml]: I reverted this change.

**Comment by pavolloffay** [comment]
I am not sure why ES is failing, but I had to add this https://github.com/jaegertracing/spark-dependencies/pull/14/files#diff-99bf6f74faf2644e1ea92ceb96f6185aR58 to dep. tests otherwise ES container didn't start and before it worked well without it. 

cc @Dogild

**Comment by jpkrohling** [comment]
I think it's safe to assume the failure isn't related to this PR, right?

**Comment by objectiser** [comment]
Yes 😄

**Comment by pavolloffay** [comment]
merge and create issue please :P

**Pull request merged**